Ladder step2

somedd opened the pull request.

somedd committed:
Add annotations

somedd committed:
Reduce depth of sources

1. 2중for문을 나누어 구현하였습니다.
2. if-else문과 guard문을 상황에 맞게 구현했습니다.

somedd committed:
Reduce sources of main function

somedd committed:
Reduce depth of drawHorizontalLines function

godrm commented:
전체적으로 많이 깔끔해졌네요!
가장 아래 부분의 로직도 꽤 긴데 하나의 기능을 담당하는 함수로 분리해보세요.

somedd committed:
Reduce depth of main loop

1.if let구문을 2중에서 1중으로 줄였습니다.
2.guard문을 보기에 더간략한 상태로 줄였습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -22,64 +20,62 @@ func generateSetOfLadder (entryVal : Int, heightVal : Int) -> [[String]] {
+            let incompleteLadder = drawLadderWithHorizontalLines(ladderSet: Ladder)
             printCompleteLadder(ladderSet: incompleteLadder)
         }
     }
```
코드 들여쓰기를 줄이도록 바꿔보라는 의미보다는
사람수 입력 / 사다리 입력 / 출력 이렇게 3개 문단으로 구분해 볼 수 있는데 각각을 함수로 분리할 수 있습니다.
이렇게 하나의 역할,기능을 함수 단위로 분리하는 것도 중요합니다.

somedd replied:
아..이해됐습니다! 감사합니다!

somedd committed:
divide functions in main

1. printLadder라는 출력함수하나로 구현하였습니다.
2. entry, height를 생성하는 함수를 구현하였습니다.

somedd committed:
Add function of checking exit-key

1. isExitKey라는 함수로 종료조건 q에 대한 부분을 구현했습니다.

somedd committed:
Add some functions for reducing depth of main function

1. prinLadder함수를 구현해 출력부분의 함수들을 하나로 구현했습니다.
2. 종료조건 q를 확인하기 위한 isExitKey함수를 구현했습니다.
3. 참가자수, 사다리 높이 갯수를 입력받는 함수를 따로 구현했습니다.

godrm commented:
함수를 분리하는 단계에 맞춰서 수고하셨습니다. 
이제 다음 단계로 각 함수들을 역할과 책임에 따라서 구조체 struct로 추상화해서 함께 묶어보세요.
